chainerx_cc/chainerx/routines/arithmetic.cc

```diff
@@ -491,4 +491,34 @@ Array Power(const Array& x1, Scalar x2) { return internal::Binary(&PowerASImpl,
 
 Array Power(Scalar x1, const Array& x2) { return internal::Binary(&PowerSAImpl, x1, x2, GetArithmeticResultDtype(x1, x2)); }
 
+Array Fmod(const Array& x1, const Array& x2) {
+    CheckEqual(x1.shape(), x2.shape());
+    Array out = Empty(x1.shape(), x1.dtype(), x1.device());
+
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<FmodKernel>(x1, x2, out);
+    }
+
+    {
+        BackwardBuilder bb{"fmod", {x1, x2}, out};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([dtype = x1.dtype()](BackwardContext& bctx) {
+                Array gx = *bctx.output_grad();
+                bctx.input_grad() = dtype == gx.dtype() ? std::move(gx) : gx.AsType(dtype);
+            });
+        }
+        if (BackwardBuilder::Target bt = bb.CreateTarget(1)) {
+            bt.Define([x1_tok = bb.RetainInput(0), x2_tok = bb.RetainInput(1), dtype = x2.dtype()](BackwardContext& bctx) {
+                const Array& x1 = bctx.GetRetainedInput(x1_tok);
+                const Array& x2 = bctx.GetRetainedInput(x2_tok);
+                Array gx = -*bctx.output_grad() * FloorDivide(x1, x2);
```

Comment:
Seems to need fix function: https://github.com/chainer/chainer/blob/11f0a8943fc3e315801cf66947670c2cf8df3d61/chainer/functions/math/fmod.py#L32

Reply:
Sorry for the delayed response @take-cheeze . I was facing some installation issues.
Thanks for the suggestion. Since `fix` isn't yet defined in ChainerX, I have implemented it using `Where`. Apart from this, I have also fixed the tests and dtype mismatch errors and the tests pass successfully now.
Could you please review?